zerver/lib/actions.py

```diff
@@ -1498,6 +1500,10 @@ def do_send_messages(messages_maybe_none: Sequence[Optional[MutableMapping[str,
             event['sender_queue_id'] = message['sender_queue_id']
         send_event(message['realm'], event, users)
 
+        if not url_embed_preview_enabled(message['message']):
```

Comment:
We don't need this check any more, just like in https://github.com/zulip/zulip/pull/14399/commits/628a25d3e17ec04c818fb948504e73a282ac4883. The links that are in `links_for_embed` will only be `whitelisted_urls` if the flag is off, or all the URLs, if the flag is on.